src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py

```diff
@@ -196,39 +208,79 @@ def update_progress(cur_loss, iteration):
     model.hybridize()
     trainer = _mx.gluon.Trainer(model.collect_params(), 'adam')
 
-    train_loss = 0.
-    for batch in loader:
-        data = _mx.gluon.utils.split_and_load(batch.data[0], 
-            ctx_list=ctx, batch_axis=0)[0]
-        label = _mx.nd.array(
-            _mx.gluon.utils.split_and_load(batch.label[0], 
-                ctx_list=ctx, batch_axis=0)[0]
-            )
-
+    train_accuracy = _mx.metric.Accuracy()
+    validation_accuracy = _mx.metric.Accuracy()
+    
+    def compute_validation_accuracy(validation_accuracy):
+        validation_loader = _SFrameClassifierIter(validation_dataset, batch_size,
+            feature_column=feature,
+            target_column=target,
+            class_to_index=class_to_index,
+            load_labels=True,
+            shuffle=True,
+            iterations=1)
+        for validation_batch in validation_loader:
+            validation_batch_data = _mx.gluon.utils.split_and_load(
+                validation_batch.data[0], ctx_list=ctx, batch_axis=0)
+            validation_batch_label = _mx.gluon.utils.split_and_load(
+                validation_batch.label[0], ctx_list=ctx, batch_axis=0)
+            validation_outputs = []
+            for x, y in zip(validation_batch_data, validation_batch_label):
+                z = model(x)
+                validation_outputs.append(z)
+            validation_accuracy.update(
+                validation_batch_label, validation_outputs)
+
+    for train_batch in train_loader:
+        train_batch_data = _mx.gluon.utils.split_and_load(
+            train_batch.data[0], ctx_list=ctx, batch_axis=0)
+        train_batch_label = _mx.gluon.utils.split_and_load(
+                train_batch.label[0], ctx_list=ctx, batch_axis=0)
+        train_outputs = []
+
+        # Inside training scope
         with _autograd.record():
-            output = model(data)
-            loss = softmax_cross_entropy(output, label)
-        loss.backward()
-        # update parameters
-        trainer.step(1)
+            for x, y in zip(train_batch_data, train_batch_label):
+                z = model(x)
+                # Computes softmax cross entropy loss.
+                loss = softmax_cross_entropy(z, y)
+                # Backpropagate the error for one iteration.
+                loss.backward()
+                train_outputs.append(z)
+            # Compute training accuracy
+            train_accuracy.update(train_batch_label, train_outputs)
+
+        compute_validation_accuracy(validation_accuracy)
```

Comment:
I believe you're still doing this after every mini-batch

Reply:
Fixed.